src/types.ts

```diff
@@ -84,5 +84,13 @@ export interface ReactCompilerConfig {
    */
   ignoreUseNoForget?: boolean,
 
-  sources?: string[] | ((filename: string) => boolean) | null
+  sources?: string[] | ((filename: string) => boolean) | null,
+
+  /*
+   * The minimum major version of React that the compiler should emit code for. If the target is 19
+   * or higher, the compiler emits direct imports of React runtime APIs needed by the compiler. On
+   * versions prior to 19, an extra runtime package react-compiler-runtime is necessary to provide
+   * a userspace approximation of runtime APIs.
+   */
+  target?:  '17' | '18' | '19'
```

Comment:
```suggestion
  target?:  '17' | '18' | '19' | (string & {})
```

Let's add `string & {}` for `target` to accept all strings but can still autocomplete for `17`, `18` and `19`. What happens if React 20 is released in the future?

Also blame React for https://github.com/facebook/react/issues/29098